src/main/java/fi/jubic/quanta/models/QuantaAuthenticator.java

```diff
@@ -0,0 +1,73 @@
+package fi.jubic.quanta.models;
+
+import fi.jubic.quanta.dao.ExternalClientDao;
+import fi.jubic.snoozy.auth.implementation.StatefulAuthenticator;
+import fi.jubic.snoozy.auth.implementation.Token;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
+
+@Singleton
+public class QuantaAuthenticator extends StatefulAuthenticator<User> {
+    private List<Token<User>> tokens;
+    private final ExternalClientDao externalClientDao;
```

Comment:
Framework integrations (like the `fi.jubic.quanta.resources` package, filters, etc.) should only depend on controllers.

Reply:
Changed to `ExternalController`